Library/Formula/ice.rb

```diff
@@ -42,170 +32,101 @@ def install
+        system "make install"
       end
     end
 
-    ENV.O2
+    inreplace "cpp/src/slice2js/Makefile" do |s|
+        s.sub! /install:/, "dontinstall:"
+    end
 
-    # what do we want to build?
-    wb = 'config src include'
-    wb += ' doc' if build.include? 'doc'
-    wb += ' demo' if build.include? 'demo'
-    inreplace "cpp/Makefile" do |s|
-      s.change_make_var! "SUBDIRS", wb
+    if build.without? "java"
+      inreplace "cpp/src/slice2java/Makefile" do |s|
+        s.sub! /install:/, "dontinstall:"
+      end
+      inreplace "cpp/src/slice2freezej/Makefile" do |s|
+        s.sub! /install:/, "dontinstall:"
+      end
     end
 
+    # Unset ICE_HOME as it interferes with the build
+    ENV.delete("ICE_HOME")
+    ENV.delete("USE_BIN_DIST")
+    ENV.delete("CPPFLAGS")
+    ENV.O2
+
     args = %W[
       prefix=#{prefix}
-      install_mandir=#{man1}
-      install_slicedir=#{share}/Ice-3.5/slice
       embedded_runpath_prefix=#{prefix}
+      USR_DIR_INSTALL=yes
       OPTIMIZE=yes
+      DB_HOME=#{libexec}
     ]
 
-    ENV["DB_HOME"] = "#{libexec}"
-
-    args << "CXXFLAGS=#{ENV.cflags} -Wall -D_REENTRANT"
-
-    # Unset ICE_HOME as it interferes with the build
-    ENV.delete('ICE_HOME')
+    #
+    # Setting this gets rid of the optimization level and the arch flags.
+    #
+    #args << "CXXFLAGS=#{ENV.cflags}"
 
     cd "cpp" do
-      system "make", *args
       system "make", "install", *args
     end
 
-    cd "php" do
-      system "make", *args
+    cd "objective-c" do
       system "make", "install", *args
     end
 
-    if build.with? "python"
-      args << "install_pythondir=#{lib}/python2.7/site-packages"
-      args << "install_libdir=#{lib}/python2.7/site-packages"
-      cd "py" do
-        system "make", *args
+    if (build.with? "java")
+      cd "java" do
         system "make", "install", *args
       end
     end
 
-    libexec.install "#{lib}/ImportKey.class"
-  end
-
-  test do
-    system "#{bin}/icebox", "--version"
+    cd "php" do
+      args << "install_phpdir=#{share}/php"
+      args << "install_libdir=#{lib}/php/extensions"
+      system "make", "install", *args
+    end
   end
 
   def caveats
     <<-EOS.undent
-      To enable IcePHP, you will need to change your php.ini
-      to load the IcePHP extension. You can do this by adding
-      IcePHP.dy to your list of extensions:
-
-          extension=#{prefix}/php/IcePHP.dy
-
-      Typical Ice PHP scripts will also expect to be able to 'require Ice.php'.
-
-      You can ensure this is possible by appending the path to
-      Ice's PHP includes to your global include_path in php.ini:
+      If you installed with Java support the IceGrid Admin application was installed.
 
-          include_path=<your-original-include-path>:#{prefix}/php
-
-      However, you can also accomplish this on a script-by-script basis
-      or via .htaccess if you so desire...
-      EOS
+      Run `brew linkapps ice` to symlink it into /Applications.
```

Comment:
You shouldn't need this line. If `brew` detects an `.app` bundle in either `prefix` or `libexec` it'll throw this dialogue automatically.